### PR TITLE
add active threads command

### PR DESCRIPTION
**Pull request by dtinth**
# Description

the server has a problem with too many threads.

last night i asked a moderator about what kind of automation would be most valuable, it is: threads pruning.

image

this PR prepares the way to go forward. start by letting moderators check how many active threads are there.

image

later PR can extend this command to provide functionality to, e.g. generate a report or prune oldest threads or more.

image

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run `pnpm format` and my code don't have any linting issues
  
  
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by buskun** [comment]
Can we have new lines separate some sections of the code? Current code is kind of hard to read.

File: src/commands/moderators/activeThreads.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  },
+  ephemeral: false,
+  execute: async (client, interaction) => {
+    if (!interaction.guild || !interaction.isChatInputCommand()) return
+    const data = (await client.rest.get(
+      Routes.guildActiveThreads(interaction.guild.id),
+    )) as RESTGetAPIGuildThreadsResult
+    const threads = (data.threads as APIThreadChannel[])
+      .slice()
+      .sort((a, b) => +(a.last_message_id || 0) - +(b.last_message_id || 0))
```
Comment (buskun) [inline src/commands/moderators/activeThreads.ts]: We should use `??` instead of `||`.

File: src/commands/moderators/activeThreads.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            label: 'Generate report (unimplemented)',
+            value: actionSet.register(
+              'generate-report',
+              async (selectInteraction) => {
+                await selectInteraction.reply({
+                  content: 'Unimplemented!',
+                  ephemeral: true,
+                })
+              },
+            ),
```
Comment (buskun) [inline src/commands/moderators/activeThreads.ts]: Can we have a TODO comment here?

File: src/commands/moderators/activeThreads.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          },
+          {
+            label: 'Prune old threads (unimplemented)',
+            value: actionSet.register('prune', async (selectInteraction) => {
+              await selectInteraction.reply({
+                content: 'Unimplemented!',
+                ephemeral: true,
+              })
+            }),
+          },
```
Comment (buskun) [inline src/commands/moderators/activeThreads.ts]: And also here

**Comment by dtinth** [comment]
@baskun added some newlines krub

File: src/commands/moderators/activeThreads.ts (hunk 1 above)
Comment (dtinth) [reply src/commands/moderators/activeThreads.ts]: yes, old habit die hard

File: src/commands/moderators/activeThreads.ts (hunk 2 above)
Comment (dtinth) [reply src/commands/moderators/activeThreads.ts]: done, thanks!

**Review by buskun — APPROVED** [review]
LGTM Kub

**Comment by dtinth** [comment]
@buskun thank you! btw thanks for helping to review code, i added you to the dev team, please check your email for invitation

**Pull request merged**